pyvista/core/utilities/geometric_sources.py

```diff
@@ -40,6 +42,59 @@
 DOUBLE_PRECISION = _vtk.vtkAlgorithm.DOUBLE_PRECISION
 
 
+@abstract_class
+class BaseSource(_vtk.vtkAlgorithm):
+    """Superclass for all sources.
+
+    Parameters
+    ----------
+    point_dtype : str, default: 'float32'
+        Set the desired output point types. It must be either 'float32' or 'float64'.
+    """
+
+    def __init__(self, point_dtype='float32'):
+        """Initialize the geometric source class."""
+        super().__init__()
```

Comment:
This class inherits from `vtkAlgorithm` and so does `vtkConeSource`.  Will calling the `__init__` method here clash with the `vtkConeSource` `__init__`, which presumably will also call the `__init__` of `vtkAlgorithm` usage?  This class seems more like a mixin to me rather than needing to inherit from `vtkAlgorithm`.  Unfortunately this makes the typing much harder for python.